### PR TITLE
Pr78

### PR DESCRIPTION
**Pull request by mchlnix**
Removing the build workflow and (hopefully) do code style test on pull requests and everything else, only when there is a tag associated with the commit.

Case of a random - non-version - tag can be seen here: https://github.com/mchlnix/SMB3-Foundry/actions/runs/513674125

If everything works with a nightly/proper release remains to be seen.

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -1,12 +1,29 @@
...
-on:
-  push:
-    tags: [ '*' ]
+on: [push, pull_request]
 
 jobs:
-  build:
-    name: "Build"
+  code_style:
+    name: Check_Codestyle
```
Comment (fortenbt) [inline .github/workflows/publish.yml]: I'd prefer to keep these names as words with spaces for consistency.

**Review by fortenbt — COMMENTED** [review]
Is the goal here to run code style checks on every push and pull request, but only build and publish to releases on tags that match those in `.ci-tag-check.py`? That's how I'm reading it. If so, I'm good with that.

File: .github/workflows/publish.yml (hunk 1 above)
Comment (mchlnix) [reply .github/workflows/publish.yml]: I forgot, that I can use strings there. I will fix that.

**Comment by mchlnix** [comment]
> Is the goal here to run code style checks on every push and pull request, but only build and publish to releases on tags that match those in `.ci-tag-check.py`? That's how I'm reading it. If so, I'm good with that.

That's exactly it.

**Pull request merged**